src/Microsoft.Azure.ServiceBus/Core/IMessageReceiver.cs

```diff
@@ -119,8 +128,8 @@ public interface IMessageReceiver : IReceiverClient
         /// <remarks>
         /// When a message is received in <see cref="ServiceBus.ReceiveMode.PeekLock"/> mode, the message is locked on the server for this
         /// receiver instance for a duration as specified during the Queue/Subscription creation (LockDuration).
-        /// If processing of the message requires longer than this duration, the lock needs to be renewed. For each renewal, the lock is renewed by
-        /// the entity's LockDuration.
+        /// If processing of the message requires longer than this duration, the lock needs to be renewed.
+        /// For each renewal, it resets the time the message is locked by the LockDuration set on the Entity.
```

Comment:
I would even add that this is **not** a guaranteed operation and can fail, resulting in message being unlocked.

Reply:
Personally, I'm not in favor of having that in the documentation. If the call `await queueClient.RenewMessageLock(lockToken)` executes successfully without any exception, then the lock must have been renewed. Else, we would throw an exception (say in scenarios where lock has already expired). Throwing the exception is already a clear indication that it did not work.

Reply:
@nemakam receiving an exception at _run-time_ is a clear indication that something didn't work. Reading API doesn't not let you know it _can_ happen. You can be running this command a thousand times and it will be successful, but on a thousand and one it will throw an exception.

Reply:
@SeanFeldman, sure it can fail. I'm not denying the possibility of failing. 
And this operation failing wouldn't result in message being unlocked. Its the other way. Message being unlocked would result in this operation failing. That's true with almost any operation that we have. Don't think we need to specify that with every operation, that it can fail.

Reply:
And moreover, the entire description of this method should make that clear. That if you need to spend more time with your message, then you need to renew the lock, else it will expire. The concept of expiration is not something that we may have to explain again in the same summary.